src/pcache.cc

```
@@ -0,0 +1,1470 @@
+  if (cache_obj->Exists(key)) {
+    uint64_t cache_len = 0;
+    cache_obj->ZCard(key, &cache_len);
+    if (cache_len + score_members.size() <= (unsigned long)zset_cache_field_num_per_key_) {
+      return cache_obj->ZAdd(key, score_members);
+    }
+
+    std::unordered_set<std::string> unique;
+    std::list<storage::ScoreMember> filtered_score_members;
+    for (auto it = score_members.rbegin(); it != score_members.rend(); ++it) {
+      if (unique.find(it->member) == unique.end()) {
+        unique.insert(it->member);
```

cppcheck

[performance] Searching before insertion is not necessary.

(stlFindInsert)